list_merge.py

```diff
@@ -1,61 +1,71 @@
 import csv
 
-# TLDs in Alexa list
-a = []
-# TLDs in TRANCO list
-t = []
-# Holder for the comboned list
-final_list = []
-# Set the desired final list length in terms of ALEXA minumum index.
+ALEXA_LIST_FILE = "lists/ALEXA-cannonical-top1m_24-07-2019.csv"
+TRANCO_LIST_FILE = "lists/TRANCO-cannonical-top1m_24-07-2019.csv"
+FINAL_LIST_FILE = "final_list.csv"
+
+# Set the desired final list length in terms of ALEXA maximum index.
+# The final list contains the union of this many top entries from both lists.
 FINAL_IDX = 10000
 
-# Open and parse the ALEXA list from local csv.
-with open(
-    "/Users/mlopatka/Downloads/lists/ALEXA-cannonical-top1m_24-07-2019.csv", "rb"
-) as f:
-    for line in f:
-        a.append(str(line, "UTF-8").rstrip().split(",")[1])
+def read_list_csv(csv_file):
+    """Read in a top site list formatted as <n>,<tld> (no headers)."""
+    list_entries = []
+    with open(csv_file, "rb") as f:
+        for line in f:
+            tld = str(line, "UTF-8").rstrip().split(",")[1]
+            list_entries.append(tld)
+    return list_entries
+
+
+def write_list_file(tld_list, list_file):
+    """Write a top site list to file.
+
+    Writes one entry per line with no index.
+    """
+    with open(list_file, "w") as f:
+        for tld in tld_list:
+            f.write(tld + "\n")
 
+
+# Holder for the combined list
+final_list = []
+
+# Open and parse the ALEXA TLD list from local csv.
+a = read_list_csv(ALEXA_LIST_FILE)
 # Truncate ALEXA list to minimum necessary elements to process.
-a = a[0 : FINAL_IDX + 1]
+a = a[0 : FINAL_IDX]
 # Save the ALEXA list into a set type for sanity checking list combination operations.
 alexa_set = set(a)
-# This should be euqal to FINAL_IDX.
-alexa_length = len(a)
-
-# Open and parse the TRANCO list from local csv.
-with open(
-    "/Users/mlopatka/Downloads/lists/TRANCO-cannonical-top1m_24-07-2019.csv", "rb"
-) as f:
-    for line in f:
-        t.append(str(line, "UTF-8").rstrip().split(",")[1])
-
-# Truncate TANCO list to minimum necessary elements to process.
-t = t[0 : FINAL_IDX + 1]
-# Save the TANCO list into a set type for sanity checking list combination operations.
+alexa_subset_length = len(a)
+
+# Open and parse the TRANCO TLD list from local csv.
+t = read_list_csv(TRANCO_LIST_FILE)
+# Truncate TRANCO list to minimum necessary elements to process.
+t = t[0 : FINAL_IDX]
+# Save the TRANCO list into a set type for sanity checking list combination operations.
 tranco_set = set(t)
+tranco_subset_length = len(t)
 
-for i in range(0, len(alexa_set)):
-    # When we have depleted the Alexa list, stop appending TRANCO elements.
-    if len(a) == 0:
-        print(len(a))
-        print(len(t))
-        break
```

Comment:
I think we just want to iterate over the subset `a` directly. The difference between `a` and `alexa_set` will only matter if `a` has repeated elements, and in this case the final list will only retain 1 copy.